test/transportSpec.js

```diff
@@ -0,0 +1,356 @@
+        data: { foo: 'bar' },
+        id: 'ee39d171-cdd5-4268-9ec8-acc349666055',
+        expiresAt: moment().add(-1, 'seconds').format(),
+      }, 'ipr')
+
+      this.params.transfer.ilp = packet
+
+      assert.deepEqual(
+        yield Transport._validateOrRejectTransfer(this.params),
+        { code: 'R01',
+          message: 'got notification of transfer with expired packet',
+          name: 'Payment Timed Out'
+        })
+
+      yield this.rejected
+    })
+  })
+
+  describe('autoFulfillCondition', function () {
+    beforeEach(function () {
+      const { packet, condition } = Transport.createPacketAndCondition({
+        destinationAmount: '1',
+        destinationAccount: 'test.example.alice.~ipr.GbLOVv3YyLo',
+        secret: Buffer.from('shh_its_a_secret', 'base64'),
+        data: { foo: 'bar' },
+        id: 'ee39d171-cdd5-4268-9ec8-acc349666055',
+        expiresAt: moment().add(1, 'seconds').format(),
+        protocol: 'ipr'
+      })
+
+      this.params = {
+        id: 'ee39d171-cdd5-4268-9ec8-acc349666055',
+        secret: Buffer.from('shh_its_a_secret', 'base64'),
+      }
+
+      this.transfer = {
+        id: 'ee39d171-cdd5-4268-9ec8-acc349666055',
+        amount: '1',
+        to: 'test.example.alice.~ipr.GbLOVv3YyLo',
+        from: 'test.example.connie',
+        executionCondition: condition,
+        ilp: packet
+      }
+
+      // detect when autofulfill promise has resolved
+      this.fulfilled = new Promise((resolve) => {
+        this.callback = resolve
+      })
+
+      this.rejected = new Promise((resolve) => {
+        this.plugin.rejectIncomingTransfer = (id, msg) => {
+          resolve(msg)
+          return Promise.resolve()
+        }
+      })
+    })
+
+    it('should call fulfillCondition on a valid incoming transfer', function * () {
+      yield Transport.listen(this.plugin, this.params, this.callback, 'ipr')
+
+      // listener returns true for debug purposes
+      const res = yield this.plugin.emitAsync('incoming_prepare', this.transfer)
+      assert.isTrue(res[0])
+
+      yield this.fulfilled
+    })
+
+    it('should reject when it generates the wrong fulfillment', function * () {
+      this.transfer.executionCondition = 'garbage'
+      yield Transport.listen(this.plugin, this.params, this.callback, 'ipr')
+
+      // listener returns false for debug purposes
+      yield this.plugin.emitAsync('incoming_prepare', this.transfer)
+      yield this.rejected
+    })
+
+    it('should reject when packet details have been changed', function * () {
+      this.transfer.ilp = Buffer.concat([this.transfer.ilp, Buffer.from('00', 'hex')])
+      yield Transport.listen(this.plugin, this.params, this.callback, 'ipr')
+
+      // listener returns false for debug purposes
+      yield this.plugin.emitAsync('incoming_prepare', this.transfer)
+      yield this.rejected
+    })
+
+    it('should pass the fulfill function to the callback', function * () {
```

Comment:
Nitpicky but could you say all the things it's actually supposed to pass to the callback?